crates/language/src/target_language.rs

```diff
@@ -369,11 +373,60 @@ impl PatternLanguage {
     }
 }
 
+#[cfg(feature = "finder")]
+fn find_ignore_files(start_path: &Path, name: &str) -> Result<Vec<PathBuf>> {
+    use path_absolutize::Absolutize;
+
+    let mut gitignore_files = Vec::new();
+
+    let mut current_path = start_path.absolutize()?.canonicalize()?;
+    while let Some(parent) = current_path.parent() {
+        let gitignore_path = parent.join(name);
+        if gitignore_path.exists() && gitignore_path.is_file() {
+            gitignore_files.push(gitignore_path);
+        }
+        current_path = parent.to_path_buf();
+    }
+
+    for entry in WalkDir::new(start_path).into_iter().filter_map(|e| e.ok()) {
+        let path = entry.path();
+        if path.file_name() == Some(std::ffi::OsStr::new(name)) {
+            gitignore_files.push(path.to_path_buf());
+        }
+    }
+
+    Ok(gitignore_files)
+}
+
+#[cfg(feature = "finder")]
+fn is_file_ignored(path: &Path) -> Result<bool> {
+    use ignore::{gitignore::GitignoreBuilder, Match};
+    use path_absolutize::Absolutize;
+
+    let git_ignores = find_ignore_files(path, ".gitignore")?;
```

Comment:
We'll want to memoize the .gitignore finding. It's very expensive to do it for every file.